### DISCUSSION TITLE
Kustomize for core immich setup

### DISCUSSION THREAD

**Discussion by SISheogorath** [immich-app/immich#7792] [Show and tell]
I'm not a big fan of these "include common"-helm charts, that are based on some popular helm app templates, since they are overly abstract.

Therefore I spend my afternoon to get rid of my own setup running on the official immich helm chart and replace it with a kustomize-based setup, that allows nicer integration of external components like operator-based PostgreSQL databases etc.

The result is a very focused, core immich installation with basically no defaults beyond the container image defaults and a reference for the machine-learning service.

https://git.shivering-isles.com/shivering-isles/infrastructure-gitops/-/tree/3d9c54bbac1be4125f8c62de90cd2e1c830afd42/apps/base/immich

This should help to fit immich into existing setups that have standardised solutions for postgresql and redis.

**Comment by bo0tzz** [comment]
> nicer integration of external components like operator-based PostgreSQL databases etc.

Note that there is nothing stopping you from doing this with the helm chart ;). That said, I do understand preferring a solution that is more straightforward.

**Reply by SISheogorath** [reply]
Yes, it's absolutely possible and it works absolutely fine. I've been running immich that way until today. I think it's mostly a question of flavour. The charts is much more batteries included.
